chainer/functions/loss/huber_loss.py

```diff
@@ -0,0 +1,40 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class HuberLoss(function.Function):
+
+    def __init__(self, delta):
+        self.delta = delta
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        type_check.expect(
+            in_types[0].dtype == numpy.float32,
+            in_types[1].dtype == numpy.float32,
+            in_types[0].shape == in_types[1].shape
+        )
+
+    def forward(self, inputs):
+        xp = cuda.get_array_module(*inputs)
+        x0, x1 = inputs
+        self.diff = x0 - x1
+        flag = xp.abs(self.diff) <= self.delta
+        y = (flag * 0.5 * xp.square(self.diff) +
+             (~flag) * self.delta * (xp.abs(self.diff) - 0.5 * self.delta))
+        y = y.astype(numpy.float32)
+        return y.sum(axis=1),
```

Comment:
> It would be made slightly faster.
y = xp.square(self.diff)
mask = y > (self.delta ** 2)
y -= mask * xp.square(abs(self.diff) - self.delta)
y *= 0.5
return y.sum(axis=1),

Is this correct? It looks different from the definition. https://en.wikipedia.org/wiki/Huber_loss
It fails on the test previously created.